component.go

```diff
@@ -58,9 +60,14 @@ type portHandler struct {
 	onClose reflect.Value
 }
 
+// inputCounter keeps track of total additions and decrements have occured
+// on inputsClose
+
+var inputCounter = 0
```

Comment:
A global counter indicates that something wrong is going on and there could be data races. What's the reason behind it?